pkg/validation/spec/header_test.go

```diff
@@ -88,3 +89,19 @@ func TestIntegrationHeader(t *testing.T) {
 
 	assertParsesJSON(t, headerJSON, header)
 }
+
+// Makes sure that a Header unmarshaled from known good JSON, and one unmarshaled
+// from generated JSON are equivalent.
+func TestHeaderSerialization(t *testing.T) {
+	generatedJSON, err := json.Marshal(header)
+	require.NoError(t, err)
+
+
+	generatedJSONActual := Header{}
+	require.NoError(t, json.Unmarshal(generatedJSON, &generatedJSONActual))
+	require.EqualValues(t, header, generatedJSONActual)
```

Comment:
can this use reflect.DeepEqual instead? various comparison libraries have disconcerting ways of coercing almost-equal values for comparison (floats/ints, null vs empty lists, etc)

Reply:
@liggitt the existing test also uses `EqualValues` from the same library here: https://github.com/kubernetes/kube-openapi/blob/cf3229a1ee8e0089b4f80d3d53ae8ecffb8b9426/pkg/validation/spec/header_test.go#L87 I expect you would like that removed as well?

A nice benefit of using this assertion method is that in the case of failure, a readable diff is shown. I am unsure if reflect has the same functionality. An alternative might be that testify has `require.Equal` as opposed to `require.EqualValues` but its documentation not detailed, and is not explicit about a promise not to coerce types.

Reply:
yeah, reflect.DeepEqual is easier to reason about... if the test passes with that, I'd prefer switching to it. If the test doesn't pass with that, then I guess I'd want to know why and whether we care about the difference that method found

Reply:
a pattern in other unit tests is to check equality with reflect.DeepEqual, then use something like cmp.Diff to print a pretty diff in failure cases

Reply:
the test now follows the pattern you suggested